app/controllers/declaration/PackageInformationController.scala

```diff
@@ -84,19 +84,31 @@ class PackageInformationController @Inject()(
         updatedCache =>
           if (updatedCache != cachedData)
             updateExportsCache(itemId, updatedCache)
-              .map(
-                _ =>
-                  navigator
-                    .continueTo(controllers.declaration.routes.CommodityMeasureController.displayPage(mode, itemId))
-              )
+              .map(_ => navigator.continueTo(nextPage(mode, itemId, request)))
           else
             Future
-              .successful(
-                navigator
-                  .continueTo(controllers.declaration.routes.CommodityMeasureController.displayPage(mode, itemId))
-            )
+              .successful(navigator.continueTo(nextPage(mode, itemId, request)))
       )
 
+  private def updateExportsCache(itemId: String, updatedCache: Seq[PackageInformation])(
+    implicit r: JourneyRequest[AnyContent]
+  ): Future[Option[ExportsDeclaration]] =
+    updateExportsDeclarationSyncDirect(model => {
+      val item: Option[ExportItem] = model.items
+        .find(item => item.id.equals(itemId))
```

Comment:
please use `updateItem` method from models - it would provide exactly this

Reply:
Done.